targets/TARGET_NUVOTON/TARGET_M480/device/startup_M480.c

```diff
@@ -426,6 +426,18 @@ void Reset_Handler_2(void)
      * Otherwise, we might get data corruption.
      */
 #if defined(FEATURE_UVISOR)
+    /* Since mbed-os-5.6.0, debug_semihosting_enable() needs to be called at the very start to 
+     * enable output of uVisor core debug message via semi-hosting. 
+     *
+     * The call must place in front of uvisor_init() because:
+     * 1) uVisor core initialization messages are output in uvisor_init().
+     * 2) The call must be in privileged mode and execution will be de-privileged on exiting uvisor_init();
+     */
+#ifdef TARGET_DEBUG
+#if defined(MBED_CONF_TARGET_UVISOR_DEBUG_SEMIHOSTING_ENABLE) && MBED_CONF_TARGET_UVISOR_DEBUG_SEMIHOSTING_ENABLE
+    uvisor_api.debug_semihosting_enable();
```

Comment:
Hi @ccli8 ,
I think you are missing the point of having `uvisor_api.debug_semihosting_enable()`.

The idea was to have a single binary that can be executed both with debugger attached and debugger detached. There is no way to reliably programmatically determine whether debugger is attached (on all ther platforms). So we added this function to prevent MCU from halting on `BKPT` instruction.
We need a debugger to be attached to get the semihosting prints. In such a case - you can add a simply call to `uvisor_api.debug_semihosting_enable()` to the debugger init script.
 
Calling `uvisor_api.debug_semihosting_enable()` function statically guarded by `ifdef` is misaligned with our goal.



Please refer to the https://github.com/ARMmbed/uvisor/blob/master/docs/lib/DEBUGGING.md

Reply:
@alzix The **NUMAKER_PFM_M487** target doesn't support GDB debugger. Currently I can debug by following the [Build with GCC and Debug with uVIsion](https://github.com/OpenNuvoton/NuMaker-mbed-uVisor-example/blob/master/BUILD_GCC_DEBUG_UVISION.md) guide. Do you know how to call `uvisor_api.debug_semihosting_enable()` in uVision debugger?

Reply:
There is field under Debugger settings tab in `uVision5` that allows specifying `Initialization file`.
`uvisor_api.debug_semihosting_enable()` can be called from it.
You will probably need to clear the `Run to main()` checkbox, otherwise initialization file commands will only be executed once you reach main.

Reply:
@alzix The approach is failed on my uVision 5.24. I try with another simple example and find C function call by debugger doesn't work whether in Initialization File, Command Window, or Watch Window.

Reply:
@alzix  I remove the commit with `uvisor_api.debug_semihosting_enable()`. If there are other means to enable uvisor semi-hosting in Keil debugger, please let me know. Thanks.